projects/components/src/modal/default-modal-ref.ts

```diff
@@ -0,0 +1,32 @@
+import { Observable, Observer, ReplaySubject } from 'rxjs';
+import { PopoverRef } from '../popover/popover-ref';
+import { ModalRef } from './modal';
+
+export class DefaultModalRef extends ModalRef<unknown> {
+  public readonly closed$: Observable<unknown>;
+
+  private readonly closedObserver: Observer<unknown>;
+  private popoverRef?: PopoverRef;
+
+  public constructor() {
+    super();
+    const closedSubject = new ReplaySubject<unknown>(1);
+    this.closedObserver = closedSubject;
+    this.closed$ = closedSubject.asObservable();
+  }
+
+  public initialize(popoverRef: PopoverRef): void {
+    this.popoverRef = popoverRef;
+    this.popoverRef.closed$.subscribe({
+      complete: () => this.closedObserver.complete(),
+      error: err => this.closedObserver.error(err)
+    });
+  }
+
+  public close(result?: unknown): void {
```

Comment:
Why is result unknown? I would think it would be either a boolean for confirmation modal or undefined for generic modals.

Reply:
because we don't care what it is in this file. the ModalRef api uses type parameters here, but since this implementation is private and used for any modal with any type, by keeping the unknown internal to the implementation, it saves us from having to stamp around `DefaultModalRef<unknown>` when working with it in the service.

Reply:
Also confirmation modal is just another modal, nothing special about it.

Reply:
Okay. Do you also plan to add a create confirmation modal method as a lot of the code over there is common and currently duplicated across many different implementations.

Reply:
yep - it's in the initial PR of this that this code is based on ( https://github.com/hypertrace/hypertrace-ui/pull/216 ), but separating the confirmation modal into its own PR.

Reply:
Cool